chemreg/substance/tests/test_views.py

```diff
@@ -578,3 +579,58 @@ def test_intermediate_user_post(client, admin_user, synonym_type_factory):
         {"data": {"type": "synonymType", "attributes": stf.initial_data}},
     )
     assert resp.status_code == 201
+
+
+def test_substance_relationship_view():
+    """Tests that the Substance Relationship View Set, is a sub class of the Model View Set."""
+    assert issubclass(SubstanceRelationshipViewSet, ModelViewSet)
+
+
+@pytest.mark.django_db
+def test_substance_relationship_list(
+    client, admin_user, substance_relationship_factory
+):
+    client.force_authenticate(user=admin_user)
+    substance_relationship_factory.create()
+    resp = client.get("/synonyms")
+    assert resp.status_code == 200
+    # Check that all results contain
+    for result in resp.data["results"]:
+        assert "qc_note" in result
+        assert "source" in result
+        assert "to_substance" in result
+        assert "from_substance" in result
+        assert "relationship_type" in result
```

Comment:
This isn't catching that the endpoint being fetched isn't correct because there are no synonyms.  As a result there are no rows in the for loop.

```suggestion
    assert resp.data["results"]
    # Check that all results contain
    for result in resp.data["results"]:
        assert "qc_notes" in result
        assert "source" in result
        assert "to_substance" in result
        assert "from_substance" in result
        assert "relationship_type" in result
```